pkg/http/handler/timeout.go

```diff
@@ -43,40 +52,33 @@ import (
 // https://golang.org/pkg/net/http/#Handler.
 //
 // The implementation is largely inspired by http.TimeoutHandler.
-func TimeToFirstByteTimeoutHandler(h http.Handler, dt time.Duration, msg string) http.Handler {
-	return &timeoutHandler{
-		handler: h,
-		body:    msg,
-		timeout: dt,
+func TimeToFirstByteTimeoutHandler(h http.Handler, msg string, timeoutFunc timeoutFunc) http.Handler {
+	return &timeToFirstByteTimeoutHandler{
+		handler:     h,
+		body:        msg,
+		timeoutFunc: timeoutFunc,
 	}
 }
 
-type timeoutHandler struct {
-	handler http.Handler
-	body    string
-	timeout time.Duration
-}
-
-func (h *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+func (tw *timeToFirstByteTimeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
just out of interest, it looks like we renamed this to `tw` which forced us to rename `tw` later on to `twriter`.. any reason not to keep this as `h`? (also since `tw` isn't an obvious shortening of `timeToFirstByteTimeoutHandler` in any way I can think of :))

Reply:
Should definitely be `h`. Missed it when refactoring from TimeoutWriter.

Reply:
👍